### ISSUE TITLE
onTimeout should run on the original thread

### ISSUE THREAD

**Issue by scruffyfox** [tom-bell/Bearing#16]
Like in onFailure for the LocationListener class, onTimeout should be called from the same thread that created it

https://github.com/tom-bell/bearing/blob/master/src/main/java/net/atomcode/bearing/location/LocationTask.java#L71

**Comment by tom-bell** [comment]
Yeah, it totally should. I'm happy to review a pull request if you want to create one?
